src/v/pandaproxy/json/iobuf.h

```diff
@@ -78,12 +80,24 @@ class rjson_serialize_impl<iobuf> {
     bool
     encode_base64(rapidjson::Writer<rapidjson::StringBuffer>& w, iobuf buf) {
         if (buf.empty()) {
-            return w.String("", 0);
+            return w.Null();
         }
         // TODO Ben: Implement custom OutputStream to prevent this linearization
         return w.String(iobuf_to_base64(buf));
     };
 
+    bool encode_json(rapidjson::Writer<rapidjson::StringBuffer>& w, iobuf buf) {
+        if (buf.empty()) {
+            return w.Null();
+        }
+        iobuf_parser p{std::move(buf)};
+        auto str = p.read_string(p.bytes_left());
+        static_assert(str.padding(), "StringStream requires null termination");
```

Comment:
this is not needed.  sstring does this in the constructor if the last char is not `\0`

Reply:
I'm asserting that the string type returned has the `NulTerminate` template argument set to true, the assert doubles as documewntation that null terminated string is required for `StringStream` constructor.

Reply:
got it.

Reply:
Also, running off the end of the string into user data is probably a security vuln, so I like the assert!